src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -921,6 +921,10 @@ public boolean isSatisfiedBy(TableMetadata metadata, DecoratedKey partitionKey,
                         ByteBuffer foundValue = getValue(metadata, partitionKey, row);
                         return foundValue != null && mapType.getSerializer().getSerializedValue(foundValue, value, mapType.getKeysType()) != null;
                     }
+                case ANALYZER_MATCHES:
+                    // The analyzer is unable to filter raw rows. It can only match analyzed columns.
+                    // Building the query restrictions should prevent reaching this code block.
+                    throw new UnsupportedOperationException("The analyzer is unable to filter raw rows. It can only match analyzed columns.");
```

Comment:
If we reach this case we have an issue in our code early on. The Assertion should handle this case but we should probably make the assertion message to make obvious what the operator was.

Reply:
Which assertion are you referencing? After fixing this PR with additional checks in the StatementRestrictions class, I haven't actually hit this condition.

Reply:
And you should never hit it ;-) I am talking about the assertion ending the switch.

Reply:
Got it, that makes sense